metadata_generation.py

```diff
@@ -0,0 +1,44 @@
+import pandas as pd
+import os
+import argparse
+
+
+def combine_metadata(
+        metadata_csv:str,
+        chexpert_csv:str,
+        negbio_csv:str,
+        split_csv:str,
+        output_dir:str,
+) -> None:
+    output_dir = os.path.join(output_dir, 'combined_metadata')
+    os.makedirs(output_dir, exist_ok=True)
+
+    chexpert=pd.read_csv(chexpert_csv, compression='gzip')
+    metadata= pd.read_csv(metadata_csv, compression='gzip')
+    negbio=pd.read_csv(negbio_csv, compression='gzip')
+    split=pd.read_csv(split_csv, compression='gzip')
+    meta_chex = metadata.merge(chexpert, on=["subject_id", "study_id"], how="left")
+    extracted_col = split["split"]
+    combine_meta_chex=meta_chex.join(extracted_col)
+    combine_meta_chex.to_csv(os.path.join(output_dir, 'combine_meta_chex.csv'), encoding='utf-8')
+    meta_neg = metadata.merge(negbio, on=["subject_id", "study_id"], how="left")
+    combine_meta_neg=meta_neg.join(extracted_col)
+    combine_meta_neg.to_csv(os.path.join(output_dir, 'combine_meta_neg.csv'), encoding='utf-8')
```

Comment:
Save output as an XLSX file with indices starting with 1 rather than 0

Reply:
XLSX doesn't open for me, So I have tested my code on CSV. It should work out for both. 
I am saving the file in xlsx format in the current version of the code.